### ISSUE TITLE
functional operations support

### ISSUE THREAD

**Issue by franklucky001** [zyedidia/generic#24]
Is there a plan to support `iterator` and common functional operations such as `map` `reduce` `filter` `scan` `fold`...

**Comment by zyedidia** [comment]
I don't currently have any plans to add iterator support, but I am happy to discuss potential designs if someone wants to try implementing it.

**Comment by rahulkhairwar** [comment]
Hi @zyedidia , if nobody is currently working on iterator support and functional operations, I will be glad to work on it!

**Comment by zyedidia** [comment]
Awesome, better iterators would be great! Would help solve issues like #37.

**Comment by rahulkhairwar** [comment]
Yup, I'm currently going through the codebase. Let me try and work up a WIP PR in the upcoming weeks!
